docs/oci-storage.md

```diff
@@ -1,40 +1,152 @@
-Fleet won't be limited by [`etcd` size limitations](https://etcd.io/docs/v3.4/dev-guide/limit/). However,
-Fleet needs write access to a registry and that registry needs to be accessible by the agents in downstream clusters.
+Fleet stores Kubernetes bundle resources in etcd twice by default. This is done using the K8s API. However, etcd has storage limits and is not optimized for large workloads. As your deployments scale, it is more efficient to store these resources externally in an OCI (Open Container Initiative) registry.
 
-## Summary
+:::note
+Fleet recommends you to compress and encode the bundle into base64 to reduce the resource size.
+:::
 
-Fleet stores by default the bundle resources in etcd twice. This is done via the k8s API and there is a size limit, depending on the etcd configuration.
+If your bundle resource is greater than X MB(size), then it would be a better alternative to switch to OCI Registry as Storage. 
 
-This feature will allow users to choose an OCI registry as storage for bundle resources. The bundle resource will have an empty resource list and a reference to the OCI repository server. The bundledeployment will not point to a content resource, but to an OCI repository server instead.
+Using an OCI registry helps you:
 
-When using this feature the bundle resources are stored once, in the configured OCI registry, and Fleet won't be tied to possible `etcd` size limitations.
+* Reduce etcd load by offloading large bundle content.  
+* Use a standardized storage backend for large manifests or Helm charts.
 
-This may be interesting for users who need to store big `Bundles`, and could also be seen as the first step for an `OCIOps` feature in the future.
+Once the OCI registry is enabled, Fleet will use it as the source for storing Bundle resources. When Fleet can't access the OCI registry, it won't fall back to default etcd storage. Instead, it logs errors.
 
-Once the OCI registry is enabled, Fleet will use it as the source for storing `Bundle` resources.
-When Fleet can't access the OCI registry, it won't fall back to default `etcd` storage. Instead, it will log errors so they can be fixed.
+![A visual asset displaying the flow of Fleet with OCI Storage.](../static/img/fleet-ociStorage-flow.png)
 
-## Configuring the OCI registry
+## **Prerequisites**
 
-OCI registry values should be configured as an extra section in the `GitRepo` yaml.
+* A running OCI registry.  
+* A Kubernetes secret with valid credentials.  
+* A Fleet installation (v2.12.0 or later) .
 
-There are the fields involved:
+## Creating an OCI Storage Secret
+
+To enable OCI Storage, you have to define a specific kind of secret. There are two ways of defining secrets
+
+* **Global secret (ocistorage)** : Applies to all GitRepos in the namespace.  
+  * This is the fallback secret. If no GitRepo-level secret is specified, Fleet uses this secret for all GitRepos in the namespace.  
```

Comment:
I think we need to mention explicitly that the name of the namespace secret is `ocistorage`. I see you are mentioning the name but I think we have to be more explicit.